packages/@okta/vuepress-site/use_cases/inline_hooks/registration_hook/registration_hook/index.md

```diff
@@ -0,0 +1,244 @@
+
+## About
+
+The Okta Registration Inline Hook allows you to integrate your own custom code into Okta’s [Self-Service Registration](https://help.okta.com/en/prod/Content/Topics/Directory/eu-self-service.htm) flow. The hook is triggered after Okta receives the registration request but before the user is created. Your custom code can:
+
+- Set or override the valuse that will be populated in attributes of the user's Okta profile
+- Allow or deny the registration attempt, based on your own validation of the information the user has submitted
+
+## Objects in the Request from Okta
+
+The outbound call from Okta to your external service includes the following objects in its JSON payload:
+
+### data.userProfile
+
+This object contains name-value pairs for each attribute supplied by the user in the Self-Service Registration form, except for password.
+
+Using the `com.okta.user.profile.update` commands you send in your response, you can modify the values of the attributes, or add values for other attributes, before the values are assigned to the Okta user profile created for the registering user.
+
+You can only set values for profile fields which already exist in your Okta user profile schema. Registration Inline Hook functionality can only set values, it cannot create new fields.
+
+> Note: The `password` field, along with any attributes that are marked as sensitive in your Okta user schema, are omitted from the information sent to your external service in the `data.userProfile` object. The password or its hash is never sent to your external service in any way.
+
+### data.action
+
+The action that Okta is currently set to take, regarding whether to allow this registration attempt.
+
+There are two possible values:
+
+- `ALLOW` indicates that the registration attempt will be allowed to proceed
+- `DENY` indicates that the registration attempt will be terminated (no user will be created in Okta)
+
+The action is `ALLOW` by default (in practice, `DENY` will never be sent to your external service).
+
+Using the `com.okta.action.update` [command](#supported-commands) in your response, you can change the action that Okta will take.
+
+## Objects in Response You Send
+
+For the Registration Inline Hook, the `commands` and `error` objects that you can return in the JSON payload of your response are defined as follows:
+
+### commands
+
+The `commands` object lets you invoke commands to modify or add values to the attributes in the Okta user profile that will be created for this user, as well as to control whether or not the registration attempt is allowed to proceed.
+
+This object is an array, allowing you to send multiple commands in your response. Each array element requires a `type` property and a `value` property. The `type` property is where you specify which of the supported commands you wish to execute, and `value` is where you supply parameters for that command.
```

Comment:
I don't know if the docs team has rules for this sort of thing, but for these first mentions of `type` and `value`, I might link to their sections below.